### PR TITLE
handle unavailable lights

### LINKED ISSUE #3540: Lovelace Light card allows interacting with unavailable lights
**Home Assistant release with the issue:**
0.98

**UI (States or Lovelace UI?):**
Lovelace

**Description of problem:**
Having a lot of recent issues with Hue lights not connecting upon restart lately, so they are often unavailable.

I have noticed that the [Lovelace Light card](https://www.home-assistant.io/lovelace/light/) still allows you to interact with the brightness slider and even displays the brightness percentage despite the light(s) being unavailable. See [screen recording](https://gfycat.com/aggravatingfirsthandazurewingedmagpie). 

I think that instead the slider shouldn't be movable / in a disabled state and maybe display "Light unavailable" in the middle where the brightness usually is too. When a light is unavailable the controls aren't shown on the more-info dialog, so it would make sense to do the same here for consistency sake and to avoid confusion.

### PR DESCRIPTION
**Pull request by iantrich**
Fixes https://github.com/home-assistant/home-assistant-polymer/issues/3540

### CODE REVIEW

**Review by balloob — APPROVED** [review]
This is _okay_ but not perfect. I think that it's better if we just make the card look disabled so that it's clear that it's not available vs not found. Now the user will not know the difference.

**Comment by iantrich** [comment]
I was having a similar thought for just the UI in general so that views could still be rendered properly but still highlight things like this

**Comment by iantrich** [comment]
I think we should either go with this or a slight change to specify it as unavailable OR we brainstorm how to better show unavailable cards. Perhaps some kind of opacity layer on the card that blocks input and conveys it as unavailable?

**Comment by balloob** [comment]
Yeah, let's block input /grey out text so it's clear that it is disabled.

As an in between solution, a yellow card with the text not available (instead of not found) is fine.

**Comment by iantrich** [comment]
@balloob I was thinking an actual overlay on the card that grey's out the overall card. That way we can have a more generic method for handling this, rather than custom solutions for each card type

**Comment by iantrich** [comment]
image

@balloob something like this

**Comment by iantrich** [comment]
If that ends up being something we want to pursue, I'll extract it to its own element and add blocking of click events on the card itself

**Comment by balloob** [comment]
that is great. However, it might be a pain for each card to be able to render without any proper information being available. Lots of if-checks. Then again, so would disabled be.

**Review by balloob — APPROVED** [review]

File: src/panels/lovelace/cards/hui-light-card.ts [hunk 1]
```diff
@@ -94,6 +95,13 @@ export class HuiLightCard extends LitElement implements LovelaceCard {
     return html`
       ${this.renderStyle()}
       <ha-card>
+        ${stateObj.state === "unavailable"
+          ? html`
+              <div id="overlay" @click="${this._stopPropagation}">
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-light-card.ts]: It's better to not use ids but use classes. So call this one the `.disabled-overlay` and target the text with `.disabled-overlay div`

**Review by balloob — APPROVED** [review]
ok to merge when final channges addressed.

**Pull request merged**